extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcTenantConfig.java

```diff
@@ -467,6 +483,59 @@ public void setPrincipalClaim(String principalClaim) {
         }
     }
 
+    @ConfigGroup
+    public static class Proxy {
+
+        /**
+         * The host (name or IP address) of the Proxy.<br/>
+         * Note: If OIDC adapter needs to use a Proxy to talk with OIDC server (Provider),
+         * then at least both "host" and "port" items must be configured to enable the usage of a Proxy.
+         */
+        @ConfigItem
+        public Optional<String> host = Optional.empty();
+
+        /**
+         * The port number of the Proxy.
+         */
+        @ConfigItem
+        public Optional<Integer> port = Optional.empty();
+
+        /**
+         * The username, if Proxy needs authentication.
+         */
+        @ConfigItem
+        public Optional<String> username = Optional.empty();
+
+        /**
+         * The password, if Proxy needs authentication.
+         */
+        @ConfigItem
+        public Optional<String> password = Optional.empty();
+
+        /**
+         * The proxy type. Default value is "HTTP".<br/>
+         * Allowed values (as defined in {@link io.vertx.core.net.ProxyType}) are: "HTTP", "SOCKS4", and "SOCKS5".
+         */
+        @ConfigItem(defaultValue = "HTTP")
+        public Optional<String> type = Optional.empty();
+
+        /**
+         * Get this proxy config as {@link ProxyOptions}
+         * 
+         * @return a ProxyOptions objects built based on the included config items
+         */
+        ProxyOptions asProxyOptions() {
```

Comment:
Please move this to the internal `OidcRecorder` helper, we don't really put `Vertx` dependencies in `OidcTenantConfig`; instead please add getters/setters (as is done alsewhere), getters are not really needed, but setters would help to `TenantConfigResolver` multi-tenancy implementations which create `OidcTenantConfig` :-)

Reply:
You are right, indeed. 👍